os/services/orchestra/orchestra-conf.h

```diff
@@ -126,32 +126,38 @@
 #define ORCHESTRA_COLLISION_FREE_HASH             0 /* Set to 1 if ORCHESTRA_LINKADDR_HASH returns unique hashes */
 #endif /* ORCHESTRA_CONF_COLLISION_FREE_HASH */
 
-/* Channel offset for the EB rule, default 0 */
-#ifdef ORCHESTRA_CONF_EB_CHANNEL_OFFSET
-#define ORCHESTRA_EB_CHANNEL_OFFSET               ORCHESTRA_CONF_EB_CHANNEL_OFFSET
-#else
-#define ORCHESTRA_EB_CHANNEL_OFFSET               0
-#endif
-
-/* Channel offset for the default common rule, default 1 */
+/* Channel offset for the default common rule, default 0 */
 #ifdef ORCHESTRA_CONF_DEFAULT_COMMON_CHANNEL_OFFSET
 #define ORCHESTRA_DEFAULT_COMMON_CHANNEL_OFFSET   ORCHESTRA_CONF_DEFAULT_COMMON_CHANNEL_OFFSET
 #else
-#define ORCHESTRA_DEFAULT_COMMON_CHANNEL_OFFSET   1
+#define ORCHESTRA_DEFAULT_COMMON_CHANNEL_OFFSET   0
 #endif
 
-/* Min channel offset for the unicast rules; the default min/max range is [2, 255] */
+/* Min channel offset for the unicast and EB rules; the default min/max range is [1, sizeof(HS)-1] if the HS has more than 1 channels and [0, sizeof(HS)-1] otherwise */
 #ifdef ORCHESTRA_CONF_UNICAST_MIN_CHANNEL_OFFSET
 #define ORCHESTRA_UNICAST_MIN_CHANNEL_OFFSET       ORCHESTRA_CONF_UNICAST_MIN_CHANNEL_OFFSET
 #else
-#define ORCHESTRA_UNICAST_MIN_CHANNEL_OFFSET       2
+#define ORCHESTRA_UNICAST_MIN_CHANNEL_OFFSET       (sizeof(TSCH_DEFAULT_HOPPING_SEQUENCE) > 1 ? 1 : 0)
 #endif
 
 /* Max channel offset for the unicast rules */
 #ifdef ORCHESTRA_CONF_UNICAST_MAX_CHANNEL_OFFSET
 #define ORCHESTRA_UNICAST_MAX_CHANNEL_OFFSET       ORCHESTRA_CONF_UNICAST_MAX_CHANNEL_OFFSET
 #else
-#define ORCHESTRA_UNICAST_MAX_CHANNEL_OFFSET       255
+#define ORCHESTRA_UNICAST_MAX_CHANNEL_OFFSET       (sizeof(TSCH_DEFAULT_HOPPING_SEQUENCE) - 1)
+#endif
+
+/* Channel offsets for the EB rule, default: same as unicast */
```

Comment:
Are we silently hurting some setups with this default?

The clearest example I can think of: If running sender-based with collison-free hash, default 4x4 hopping sequence, and have set `ORCHESTRA_UNICAST_MAX_CHANNEL_OFFSET` to e.g. 3. Would I not be getting increased EB and unicast collisions after this PR? Previously I would have EB on offset 0, common on 1, and unicast on 2-3. After the PR, EB and unicast would share 1-3.

Reply:
The example is true for networks of any size no? As long as the the hash is still collision free?

Reply:
Previously get_node_channel_offset could return something like 4, even if the hash is collision free, and 4 mod 4 is just zero, which is the same as ORCHESTRA_EB_CHANNEL_OFFSET default value.

Reply:
Yes, under these assumptions the collisions will be worse after the PR.

I still think that it's better to apply it than not, because by default the max unicast channel offset is set to 255, as said in the first message, and I suspect very few users change this default.

Reply:
Is it possible to get the best of both worlds by defaulting the EB min/max to [1, 1], and have unicast at [2, MAX(2, (HS-1)]?

Reply:
I think that would be better, yes. What do you think about the situation when we have just 2 channels? Then have 0 for default common channel offset, and 1 for both EB and unicast - does that sound the best?

Reply:
Finding the definitive answer seems like a miniature research project :) I think your suggestion is good, and should regardless be better than the existing default.

Reply:
Updated the PR. For both corner cases (there are 1 or 2 channels in channel hopping sequence) I set the unicast channel offset to 1, to simplify the macro. In case of 1 channel if will be effectively identical to 0 anyway.

Reply:
:+1: